Core/src/Material/SurfaceMaterialMapper.cpp

```diff
@@ -175,9 +176,12 @@ void Acts::SurfaceMaterialMapper::finalizeMaps(State& mState) const {
 
 void Acts::SurfaceMaterialMapper::mapMaterialTrack(
     State& mState, RecordedMaterialTrack& mTrack) const {
+  using VectorHelpers::makeVector4;
+
   // Neutral curvilinear parameters
-  NeutralCurvilinearTrackParameters start(std::nullopt, mTrack.first.first,
-                                          mTrack.first.second, 0.);
+  NeutralCurvilinearTrackParameters start(makeVector4(mTrack.first.first, 0),
+                                          mTrack.first.second,
+                                          1 / mTrack.first.second.norm());
```

Comment:
So, there's no Vector3D position constructor anymore for NeutralCurvilinear?
I think it's ok, though we will often run into situation where the time is just ignored (or forced to be set to 0).

Reply:
Apart from the deprecated ones, no. There were only a few use cases in the Core (mostly material mapping) which are already switched to the new constructors. Many of these cases could be simplified by not storing position, momentum separately at all, but to store the complete free parameters vector instead.

Reply:
Do you want me to do the cleanup of deprecated constructors today? It's worth waiting v01.00.00 for.

Reply:
I already have the commits for that but kept them out of this PR. I will put out a second PR as soon as I fixed the build errors.